### PR TITLE
Scalability framework: Allow multiple targets to be benchmarked together

### PR DESCRIPTION
**Pull request by philip-stoev**
```
commit 3497c06a0581cf7df87f63e6fb51cf4f2314d96d (HEAD -> scalability-multiple-together, origin/scalability-multiple-together)
Author: Philip Stoev <pstoev@materialize.com>
Date:   Thu Aug 24 10:14:02 2023 +0200

    scalability-framework: Remove matplotlib ipympl
    
    The ipympl engine does not update when a different workload
    is chosen from the drop-down. Revert to the default Jupyter rendering
    engine.

commit 85bf6d405b363cc5d50896d8b821bd219c22cefb
Author: Philip Stoev <pstoev@materialize.com>
Date:   Thu Aug 24 10:13:33 2023 +0200

    scalability-framework: Allow multiple targets to be benchmarked together
```

### Motivation

Previously, one had to run mzcompose twice in order to get multiple targets onto the same set of charts.

### CODE REVIEW

File: test/scalability/lib.py [hunk 1]
```diff
@@ -21,9 +21,17 @@ def plotit(csv_file_name: str) -> None:
...
-            ["git", "log", "-1", "--pretty=format:%s", target_sha], text=True
-        )
+
+        target_comment = None
+        try:
+            target_comment = subprocess.check_output(
+                ["git", "log", "-1", "--pretty=format:%s", target_sha], text=True
+            )
+        except subprocess.CalledProcessError:
+            # Sometimes mz_version() will report a Git SHA that is not available
```
Comment (nrainer-materialize) [inline test/scalability/lib.py:31]: Should we log this? (optional)

File: test/scalability/mzcompose.py [hunk 2]
```diff
@@ -222,31 +223,40 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
+    endpoints: list[Endpoint] = []
+    for i, target in enumerate(args.target):
+        endpoint: Optional[Endpoint] = None
+        if target == "local":
+            endpoint = MaterializeLocal()
+        if target == "remote":
+            endpoint = MaterializeRemote(materialize_url=args.materialize_url[i])
+        elif target == "postgres":
+            endpoint = PostgresContainer(composition=c)
+        elif args.target == "HEAD" or args.target is None:
```
Comment (nrainer-materialize) [inline test/scalability/mzcompose.py:251]: Hm... having `args.target` with a `None` value felt okay when it was a single value. Now, it is a list and I don't like `[None]` (line 240) that much. I would prefer a special string constant (`"NOT_SPECIFIED"`) instead to reduce magic.

**Review by nrainer-materialize — APPROVED** [review]
Cool feature extension 👍

File: test/scalability/mzcompose.py (hunk 2 above)
Comment (philip-stoev) [reply test/scalability/mzcompose.py:251]: So now it is a list and yet the type checker allows me to compare it to a string :-( This entire line is wrong now , will fix.

File: test/scalability/lib.py (hunk 1 above)
Comment (philip-stoev) [reply test/scalability/lib.py:31]: Yes, it is visible in the Jupyter notebook when it happens -- the original error from git log is shown in the notebook. I am still at a loss why it happens so frequently, so will need to figure out an alternative way to obtain the git comment going forward.

File: test/scalability/mzcompose.py (hunk 2 above)
Comment (philip-stoev) [reply test/scalability/mzcompose.py]: Thanks for spotting this. Fixed.

**Pull request merged**